src/Polly.Core/Simmy/Outcomes/OutcomeGenerator.cs

```diff
@@ -35,6 +35,9 @@ internal OutcomeGenerator(Func<int, int> weightGenerator)
     /// <param name="generator">The delegate that generates the exception.</param>
     /// <param name="weight">The weight assigned to this generator. Defaults to <c>100</c>.</param>
     /// <returns>The current instance of <see cref="OutcomeGenerator{TResult}"/>.</returns>
+#pragma warning disable S1133 // Deprecated code should be removed
+    [Obsolete("This method is deprecated and will be removed in the next version. Use Chaos fault strategy instead.")]
+#pragma warning restore S1133 // Deprecated code should be removed
     public OutcomeGenerator<TResult> AddException(Func<Exception> generator, int weight = DefaultWeight)
```

Comment:
Why obsolete this though? I consider this still a valid pattern. Imagine a very common scenario where we need to inject both types of faults `HttpResponseMessage` with 500 and `HttpRequestException`.

Current chaos outcome API is powerful enough to handle both cases using a single strategy. My thinking for anti-patters was using chaos outcome to inject exceptions only and no results. In that case, the chaos fault is a better option.

Reply:
The scenario you're talking about makes a lot of sense, but still, both have undesirable side effects:  telemetry and control. We could certainly modify the outcome strategy to report the exception in a different way, but to me we're mixing two different things, exceptions and "faulted" results, since the last one isn't necessary an exception.

On the other hand the biggest downside of that approach is the consumer is losing control, in the real world you usually have different configurations to inject different kinds of chaos, it does not feel right treating outcomes the same as exceptions, so imagine you have a 10% as injection rate for outcomes and 1% as injection rate for exceptions, if we allow injecting exceptions via the outcome strategy it will completely disregard the fault configuration.

Reply:
My biggest problem here is obsoleting API that was introduced just one version ago. In Polly v8, the `Outcome` can represents both results and exceptions. Even the signature of chaos outcome allows this:

https://github.com/App-vNext/Polly/blob/de6eb20fd8b808a3f821fc2ab1eef2eeb0b99fa7/src/Polly.Core/Simmy/Outcomes/ChaosOutcomeStrategyOptions.cs#L31C4-L31C104

So at this point, I gravity toward not breaking anything. If someone has a scenario where they need both results and exceptions injected in a single strategy, let's not prevent them. (we actually have some teams that are doing this)
Let's not forget that every additional strategy adds a slight overhead, so having the option to express results and exceptions with a single strategy is beneficial.

>  it does not feel right treating outcomes the same as exceptions, so imagine you have a 10% as injection rate for outcomes and 1% as injection rate for exceptions, if we allow injecting exceptions via the outcome strategy it will completely disregard the fault configuration.

The `OutcomeGenerator` API is flexible enough to allow assigning weight to an exception, so the above can still be expressed.

``` csharp
new OutcomeGenerator<HttpResponseMessage>()
    .AddResult(() => new HttpResponseMessage(System.Net.HttpStatusCode.InternalServerError), weight: 50)
    .AddException(() => new HttpRequestException(), weight: 2);
```

Reply:
I agree that obsoleting something immediately doesn't look great. If we did, it wouldn't be in the _next_ version, but a _future_ version.

Reply:
>Even the signature of chaos outcome allows this:
> 
> https://github.com/App-vNext/Polly/blob/de6eb20fd8b808a3f821fc2ab1eef2eeb0b99fa7/src/Polly.Core/Simmy/Outcomes/ChaosOutcomeStrategyOptions.cs#L31C4-L31C104

Correct, that's why the documentation states the antipattern may occur in advanced scenarios when the use of generator delegates is required.

> The OutcomeGenerator API is flexible enough to allow assigning weight to an exception, so the above can still be expressed.

Yeah partially I guess, because that's still injecting an exception using the outcome configuration, how about if the "knob" to inject exceptions is off? As I said in the docs this is very opinionated, but it's good having the discussion so that we can enrich the docs. And to be clear, I'm not against the AddException API, it's just that I feel we need to be cautious with that and not encourage people to use it w/o knowing the caveats, I decided to obsolete it based on the premise that treating faults as outcomes is an antipattern because of the reasons I'm explaining in this PR, mostly because of:
* side effects on telemetry events, which might affect instrumentation and metrics
* goes against controlled chaos injection

Said that, other than documenting the scenarios we can also document that method using the `remarks` or something like that.